src/functional/ffront/past_passes/type_deduction.py

```diff
@@ -93,13 +93,12 @@ def visit_Program(self, node: past.Program, **kwargs):
         definition_type = ct.FunctionType(
             args=[param.type for param in params], kwargs={}, returns=ct.VoidType()
         )
-
         return past.Program(
             id=self.visit(node.id, **kwargs),
             type=ct.ProgramType(definition=definition_type),
             params=params,
             body=self.visit(node.body, **kwargs),
-            closure_vars=self.visit(node.closure_vars, **kwargs),
+            closure_vars=node.closure_vars,
```

Comment:
```suggestion
            closure_vars=self.visit(node.closure_vars, **kwargs),
```
Unless there is a reason not to, always visit the child nodes.

Reply:
Done